ion-schema/src/system.rs

```diff
@@ -948,10 +948,26 @@ impl Resolver {
             );
         }
 
-        Ok(Rc::new(Schema::new(
-            isl.id(),
-            Rc::new(type_store.to_owned()),
-        )))
+        let schema = Arc::new(Schema::new(isl.id(), Arc::new(type_store.to_owned())));
+
+        // add schema to schema cache
+        // if we are loading an import of the schema then we can only add this schema to cache if its a full schema import
+        // and can not add it to cache if we are loading specific type imports from the schema.
```

Comment:
Because when doing  a type import we are only importing specific types rather then the entire schema. So this only adds the schemas which are fully loaded into the schema cache.

Reply:
Okay, so suppose I have a schema `foo.isl`, and I only ever use types from that schema file by using type imports (`{id: "foo.isl", type: foo_bar }`), does this mean that for each other schema that uses `{id: "foo.isl", type: foo_bar }`, we need to load `foo.isl` from disk again?

Reply:
Yes, that is what this is doing right now. Currently type imports only imports the specified types, which is why there isn't really a resolved schema for it that I can store inside the cache. I can either load all types of schema every time there is a type import or I can give another name/id to this schema which has only specific types imported and then add it to cache. Another option would be to maintain a new cache for these type imports.

Reply:
I think probably maintaining another cache for it would be a better solution here. It doesn't require having any temporary name/id to schema(which might be used by another schema) and it doesn't require resolving the entire schema for each type import. 
What do you think?

Reply:
I think it should probably cache the whole schema so that we don't need to reload the schema every time a type from the schema is imported somewhere.

Also, we should probably load the entire schema because (at least in the Kotlin impl) if any part of the schema document is invalid ISL, then the whole schema document is considered invalid.